### PR TITLE
Ws5.3.1 add species controller

### PR DESCRIPTION
**Pull request by RuksanEmirali**
Created SpeciesController file
Created SpeciesService file
Created service to get list of species - GetAllSpecies.
Create SpeciesResponse model which returns the list of species and a count of species.
In controller, called the service to get a list of species. Returns Ok with SpeciesResponse or bad request with error message.
Updated Program.cs to register service. 
Checked using swagger.
image

### CODE REVIEW

File: backend/Services/SpeciesServices.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using WhaleSpotting.Models.Data;
+using WhaleSpotting.Models.Request;
+
+namespace WhaleSpotting.Services;
+
+public interface ISpeciesServices
```
Comment (Kiran-R-K) [inline backend/Services/SpeciesServices.cs:6]: As above for convention this should be `ISpeciesService`

File: backend/Controllers/SpeciesController.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        _service = service;
+    }
+
+    [HttpGet("")]
+    public IActionResult GetAllSpecies()
+    {
+        try
+        {
+            List<Species> listOfSpecies = _service.GetAllSpecies();
+            return Ok(new SpeciesResponse { ListOfSpecies = listOfSpecies, Count = listOfSpecies.Count });
```
Comment (Kiran-R-K) [inline backend/Controllers/SpeciesController.cs:26]: The `SpeciesResponse` should be being returned by the `GetAllSpecies()` method in the service. The controller should only be responsible for returning the response in the API, ideally it shouldn't be where the response is being made

Comment (Kiran-R-K) [reply backend/Controllers/SpeciesController.cs:26]: Thinking about this more, we may want to have a `GetAllSpeciesResponse` in the service that calls `GetAllSpecies()` and saves the results into a response model?

**Review by Kiran-R-K — CHANGES_REQUESTED** [review]
Nice job, I've left a couple of questions and requested a few changes, let me know if you have any questions

File: backend/Services/SpeciesServices.cs
[the remark this answers is not in the document]
Comment (RuksanEmirali) [reply backend/Services/SpeciesServices.cs:1]: Change made

File: backend/Services/SpeciesServices.cs (hunk 1 above)
Comment (RuksanEmirali) [reply backend/Services/SpeciesServices.cs]: Change made

File: backend/Services/SpeciesServices.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public List<Species> GetAllSpecies();
+}
+
+public class SpeciesServices : ISpeciesServices
```
[the remark this answers is not in the document]
Comment (RuksanEmirali) [reply backend/Services/SpeciesServices.cs]: Change made

**Review by Kiran-R-K — APPROVED** [review]
nice, really good job

**Pull request merged**